src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -85,7 +85,10 @@ protected ApiConfiguration Configuration
                     apiType =>
                     {
                         IServiceCollection services = new ServiceCollection()
-                            .CutoffPrevious<IQueryExecutor>(DefaultQueryExecutor.Instance);
+                            .CutoffPrevious<IQueryExecutor>(DefaultQueryExecutor.Instance)
+                            .AddSingleton<ApiConfigurationExtensions.PropertyBag>()
+                            .AddScoped<ApiContextExtensions.PropertyBag>()
+                            .AddScoped<InvocationContextExtensions.PropertyBag>();
```

Comment:
Got it! Does it mean that even if we declared `PropertyBag` as scoped then in `ApiConfiguration` we can get one from the root scope which is actually the same as singleton?